File: bnk_proj/lib/voice/core/voice_res_dto.dart
import 'package:test_main/voice/core/voice_intent.dart';
import 'package:test_main/voice/core/voice_state.dart';

import 'end_reason.dart';

class VoiceResDTO {
  final VoiceState currentState;
  final Intent intent;
  final String? noticeCode;
  final EndReason? endReason;
  final String? productCode;


  final String? inputField;
  final String? inputValue;



  VoiceResDTO({
    required this.currentState,
    required this.intent,
    this.noticeCode,
    this.endReason,
    this.productCode,
    this.inputField,
    this.inputValue
  });

  factory VoiceResDTO.fromJson(Map<String, dynamic> json) {
    return VoiceResDTO(
      currentState: VoiceState.from(json['currentState']),
      intent: Intent.from(json['intent']),
      noticeCode: json['noticeCode'],
      endReason: json['endReason'] != null
          ? EndReason.from(json['endReason'])
          : null,
      productCode: json['productCode'],
      inputField: json['inputField'],
      inputValue: json['inputValue'],
    );
  }
}

File: backend/src/main/java/kr/co/api/backend/voice/dto/VoiceResDTO.java
package kr.co.api.backend.voice.dto;

import kr.co.api.backend.voice.domain.EndReason;
import kr.co.api.backend.voice.domain.VoiceIntent;
import kr.co.api.backend.voice.domain.VoiceState;
import lombok.Data;

@Data
public class VoiceResDTO {
    private VoiceIntent intent;
    private VoiceState currentState;
    private EndReason endReason;

    // 프론트 스크립트 매핑용 (재요청/상품필요/오류 등)
    private String noticeCode;
    private String noticeMessage;

    private String inputField;
    private String inputValue;

    // (선택) 디버깅용
    private String productCode;


}
